Mathlib/Analysis/Analytic/Composition.lean

```diff
@@ -892,6 +893,80 @@ lemma AnalyticOnNhd.comp_analyticOn {f : F → G} {g : E → F} {s : Set F}
     AnalyticOn 𝕜 (f ∘ g) t :=
   fun x m ↦ (hf _ (h m)).comp_analyticWithinAt (hg x m)
 
+/-- If two functions `g` and `f` have finite power series `q` and `p` respectively at `f x` and `x`,
+then `g ∘ f` admits the finite power series `q.comp p` at `x`. -/
+theorem HasFiniteFPowerSeriesAt.comp {m n : ℕ} {g : F → G} {f : E → F}
+    {q : FormalMultilinearSeries 𝕜 F G} {p : FormalMultilinearSeries 𝕜 E F} {x : E}
+    (hg : HasFiniteFPowerSeriesAt g q (f x) m) (hf : HasFiniteFPowerSeriesAt f p x n) (hn : 0 < n) :
+    HasFiniteFPowerSeriesAt (g ∘ f) (q.comp p) x (m * n) := by
+  rcases hg.hasFPowerSeriesAt.comp hf.hasFPowerSeriesAt with ⟨r, hr⟩
+  refine ⟨r, hr, fun i hi ↦ ?_⟩
+  apply Finset.sum_eq_zero
+  rintro c -
+  ext v
+  simp only [compAlongComposition_apply, ContinuousMultilinearMap.zero_apply]
+  rcases le_or_gt m c.length with hc | hc
+  · simp [hg.finite _ hc]
+  obtain ⟨j, hj⟩ : ∃ j, n ≤ c.blocksFun j := by
+    contrapose! hi
+    rw [← c.sum_blocksFun]
+    rcases eq_zero_or_pos c.length with h'c | h'c
+    · have : ∑ j : Fin c.length, c.blocksFun j = 0 := by
+        apply Finset.sum_eq_zero (fun j hj ↦ ?_)
+        have := j.2
+        omega
```

Comment:
Can `cutsat` solve this, or even `cutsav [j.2]`?

Reply:
`cutsat [j.2]` is not accepted as a syntax. `cutsat` also fails without the `have`. I switched to `grind` because, if I understand correctly, `omega` is supposed to be replaced by `grind` in the future. I haven't used `cutsat` because I prefer the approach "one tactic to rule them all" which is easier to learn.